test/e2e/scheduling/limit_range.go

```diff
@@ -47,9 +51,34 @@ var _ = SIGDescribe("LimitRange", func() {
 			min, max,
 			defaultLimit, defaultRequest,
 			maxLimitRequestRatio)
-		limitRange, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).Create(limitRange)
+
+		By("Setting up watch")
+		selector := labels.SelectorFromSet(labels.Set(map[string]string{"name": limitRange.Name}))
+		options := metav1.ListOptions{LabelSelector: selector.String()}
+		limitRanges, err := f.ClientSet.CoreV1().LimitRanges(f.Namespace.Name).List(options)
+		Expect(err).NotTo(HaveOccurred(), "failed to query for limitRanges")
+		Expect(len(limitRanges.Items)).To(Equal(0))
```

Comment:
I would warn here b/c we've seen a number of test conditions fail on loaded clusters due to waiting on cleanup conditions.  I realize that it shouldn't occur, but we've seen patterns of odd cleanup when running (N) overlapping test cases with namespaces names that are not completely unique.

Reply:
What can you suggest to do to avoid the problems here?